docs/llm_tools_invocation.md

```diff
@@ -0,0 +1,90 @@
+---
+title: "LLM Tools Invocation"
+---
+
+If you are developing an LLM integration and want to add support of LLM Tools invocation, then here are the steps to do that.
+
+## Step 0. Register the `intent` component.
+
+The intents are also automatically registered as LLM tools. However in order to do so, the `intent` component must be loaded. It does not happen automatically, so it is recommended to make sure it is loaded before using the tools for the first time.
+
+```python
+from homeassistant import setup
+...
+        if "intent" not in self.hass.config.components:
+            await setup.async_setup_component(self.hass, "intent", {})
+```
+
+## Step 1. Get available tools
```

Comment:
Rather than making this a "How to" think of this as documenting the contracts and responsibilities in the APIs.  The documentation in step 2 example below is sufficient for getting tools and formatting them so I don't think it needs to have two sections and this one can be removed.